x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/utils.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { flattenHit } from '@kbn/data-service';
+import type { DataView } from '@kbn/data-views-plugin/public';
+import type { DataTableRecord } from '@kbn/discover-utils/types';
+import type { TimelineItem } from '../../../../../common/search_strategy';
+
+interface TransformTimelineItemToUnifiedRows {
+  events: TimelineItem[];
+  dataView: DataView;
+}
+
+export function transformTimelineItemToUnifiedRows(
+  args: TransformTimelineItemToUnifiedRows
+): Array<DataTableRecord & TimelineItem> {
+  const { events, dataView } = args;
+  const unifiedDataTableRows = events.map(({ _id, _index, ecs, data }) => {
+    const _source = ecs as unknown as Record<string, unknown>;
+    const hit = { _id, _index: String(_index), _source };
+    /*
+     * Ideally for unified data table we only need raw and flattened keys
```

Comment:
I think there may be some fields (nested, geo, etc..) that may not work properly without this even in the unified data table, but need to test that out because the rationale behind why we have so many formats isn't readily available